### DISCUSSION TITLE
2310.2 built from source on Ubuntu 23.10 only works after manual python symlink

### DISCUSSION THREAD

**Discussion by lucasw** [o3de/o3de#17649] [Q&A] [answered]
Checked out this version

```
commit e22213ea946932dc2f94d9f1392e98b4bf2b2c3f (HEAD, tag: 2310.2, origin/main)
Merge: c89678d6ba 06830fec9c
Author: Alex Peterson <26804013+AMZN-alexpete@users.noreply.github.com>
Date:   Wed Jan 17 11:18:38 2024 -0800

    Merge pull request #17386 from o3de/point-release/23102
    
    Point release/23102
```

I did a build from source following https://www.docs.o3de.org/docs/welcome-guide/setup/setup-from-github/building-linux/

```
sudo apt install libxcb-xkb-dev libxcb-xfixes0-dev libxkbcommon-x11-dev  # must have already had anything else needed
cmake -B build/linux -S ../../o3de/ -G "Ninja Multi-Config"
cmake --build build/linux --target Editor --config profile -j4  # 842 build steps, takes up 30-40 GB of disk for build artifacts
```

I then entered the venv that was automatically created, and tried to run it (I didn't see this in the build instructions but it seemed like the right thing)

```
source ~/.o3de/3rdParty/venv/.../bin/activate
./build/linux/bin/profile/o3de
```

A box popped up with 

![o3de Screenshot from 2024-03-25 09-29-58](https://github.com/o3de/o3de/assets/1334122/e74e1a18-0443-47e2-a3e4-fd778bbc500a)

```
This tool requires an O3DE engine with a Python runtime, but either Python is missing or mis-configured.

Press 'OK' to run the python/get_python.sh script automatically, or 'Cancel' if you want to manually resolve the issue by renaming your python/runtime folder and running python/get_python.sh yourself.
```

Pressing OK didn't work, it goes to a pop with a progress bar at 100% but never exits

![o3de Screenshot from 2024-03-25 09-24-44](https://github.com/o3de/o3de/assets/1334122/aadece7e-e4b2-47ee-818f-6ae6cda77241)

I then tried the manual get_python.sh (but maybe didn't need to do this)

```
(...) ...~/other/src/sim/build/o3de$ ../../o3de/python/get_python.sh 
...
Installing collected packages: o3de
  Attempting uninstall: o3de
    Found existing installation: o3de 1.0.0
    Uninstalling o3de-1.0.0:
      Successfully uninstalled o3de-1.0.0
  Running setup.py develop for o3de
Successfully installed o3de-1.0.0
~/other/src/sim/o3de/python
```

That appeared to conclude successfully, but the `o3de` still reported no python, but this error output was useful
```
python: Trace::Error
 /home/lucasw/other/src/sim/o3de/Code/Tools/ProjectManager/Source/PythonBindings.cpp(290): 'virtual bool O3DE::ProjectManager::PythonBindings::StartPython()'
python: Python home path does not exist: /home/lucasw/other/src/sim/build/o3de/build/linux/bin/profile/python/runtime/python-3.10.5-rev4-linux/python
```

so I symlinked ~/other/src/sim/o3de/python into build/linux/bin/profile

```
~/other/src/sim/build/o3de/build/linux/bin/profile$ ln -s ~/other/src/sim/o3de/python
```

Now I can get this far:

![Screenshot from 2024-03-25 09-32-44](https://github.com/o3de/o3de/assets/1334122/1c6f804f-1b7b-419e-a3ae-f15fd9c4bfa9)

**Comment by lucasw** [comment]
I tried the 2310.2 .deb and it also shows the get_python popup window, but pressing the OK button does work and o3de appears to be running normally after that extra install finishes.  The snap installed o3de on the other hand seems to have python configured in it properly as installed, and launches straight into the project manager on the first run.

**Reply by spham-amzn** [reply, marked as answer]
This PR: https://github.com/o3de/o3de/pull/17623 should fix this issue
